File: CinemaBooking/Screening.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace CinemaBooking
{
    internal class Screening
    {
        [JsonPropertyName("id")]
        public long Id { get; set; }


        [JsonPropertyName("movie")]
        public Movie Movie { get; set; } = new();


        [JsonPropertyName("hall")]
        public string Hall { get; set; } = "";


        [JsonPropertyName("time")]
        public DateTime Time { get; set; }


        [JsonPropertyName("seats_total")] 
        public int SeatsTotal { get; set; }


        [JsonPropertyName("seats_sold")] 
        public int SeatsSold { get; set; }

        [JsonPropertyName("status")]
        public string? Status { get; set; }
    }
}

File: Server/cinemabooking/cinemabooking/src/main/java/com/cinemabooking/db/Screening.java
package com.cinemabooking.db;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name="screening")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Screening {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional=false)
    @JoinColumn(name="movie_id")
    private Movie movie;

    @NotBlank
    private String hall;

    @NotNull
    private LocalDateTime time;

    @Positive
    private int seats_total=180;

    @Min(0)
    private int seats_sold=0;

    @Enumerated(EnumType.STRING)
    private ScreeningStatus status;
}
